### ISSUE TITLE
get_game_variable() vs state.game_variables

### ISSUE THREAD

**Issue by pengsun** [mwydmuch/ViZDoom#342]
Hi, I noticed there are two ways to get game variables,

One is based on firstly `game.add_available_game_variables`, then retrieving them by `game.get_state().game_variables`. This way, all the required variables must be specified in cfg file or python code, otherwise the `game_variables` field is `None`.

The other way is by means of `game.get_game_variable(...)`, which does not need specifying what variables are required beforehand -- simply all of them can be accessed.

Both ways exist in the `examples/python`. Wondering if there are any difference, particularly for efficiency? i.e., will the first parsimonious way cost less time when stepping each frame?

**Comment by Miffyli** [comment]
I believe there is no significant performance difference between these two when you count in the time you need for e.g. processing images.

`GameState.game_variables` makes it convenient to feed a big bunch of game variables to your model/network/agent/whatever, and you can also share the specific config of used variables in the config file. This way other people do not have to start adding extra lines of code to replicate your results or to run your environment.

However sometimes you want information for e.g. debugging or plotting your agent's movement, but you do not want to give this info to your agent. In this case `DoomGame.get_game_variable` acts as a nifty way of fetching this info anywhere in the code without needing to touch the config files.

tl;dr: They both serve their own purpose.

**Comment by pengsun** [comment]
I see! Thanks for the detailed explanation, very clear!
